qiskit/chemistry/drivers/fcidumpd/dumper.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""FCIDump dumper."""
+
+from typing import List, Optional
+from io import TextIOWrapper
+import itertools
+import numpy as np
+
+
+def dump(outpath: str, norb: int, nelec: int, hijs: List[float], hijkls: List[float], einact: float,
+         ms2: Optional[int] = 0, orbsym: Optional[List[int]] = None, isym: Optional[int] = 1
```

Comment:
The type `Optional` is only used if a value has the default `None` in the signature, not for all values that have a default value :slightly_smiling_face: This should also be adjusted in other signatures.

```suggestion
         ms2: int = 0, orbsym: Optional[List[int]] = None, isym: int = 1
```